dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonHelpers.java

```diff
@@ -228,4 +236,90 @@ protected static String jsonifyTags(String[] tags) {
         return sb.toString();
     }
 
+    protected static boolean checkExtSuffixValidity(KeyValuePair kvp) {
+        boolean isValidExt = false;
+        String key = kvp.getKey();
+        String value = kvp.getValue();
+
+        if (key.endsWith(INT_SUFFIX)) {
+            isValidExt = verifyIntValue(value);
+        } else if (key.endsWith(FLOAT_SUFFIX)) {
+            isValidExt = verifyFloatValue(value);
+        } else if (key.endsWith(BOOL_SUFFIX)) {
+            if (value.equals(TRUE_BOOL) || value.equals(FALSE_BOOL)) {
+                isValidExt = true;
+            }
+        } else {
+            isValidExt = true;
+        }
+        return isValidExt;
+    }
+
+    public static boolean verifyIntValue(String value) {
+        boolean isValid = true;
+        char[] arr = value.toCharArray();
+        for (int i = 0; i < arr.length; i++) {
+            if (i == 0) {
+                if (arr[i] == '-' && arr.length > 1) {
+                    continue;
+                }
+            }
+            if (arr[i] >= '0' && arr[i] <= '9') {
+                continue;
+            } else {
+                isValid = false;
+                break;
+            }
+        }
+        return isValid;
+    }
+
+    public static boolean verifyFloatValue(String s) {
+        boolean isValid = true;
+        boolean decimalFlag = false;
+        char[] arr = s.toCharArray();
+        for (int i = 0; i < arr.length; i++) {
+            if (i == 0) {
+                // If the string has more than 1 characters, and the first position in the string has a -, then it is okay.
+                if (arr[i] == '-' && arr.length > 1) {
+                    continue;
+                }
+            }
+            // If a '.' is found
+            if (arr[i] == '.') {
+                // If the decimal is not in the first spot, not in the last spot and the onlt decimal found
+                if (i > 0 && i < arr.length - 1 && decimalFlag == false) {
+                    // Check if there is a digit before the decimal
+                    if (arr[i - 1] >= '0' && arr[i - 1] <= '9') {
+                        // Set decimal flag
+                        decimalFlag = true;
+                        continue;
+                    } else {
+                        return false;
+                    }
+                } else {
+                    //if the decimal conditions are violated, then the number is invalid
+                    return false;
+                }
+            }
+            // Check if the current character is a digit
+            if (arr[i] >= '0' && arr[i] <= '9') {
+                continue;
+            } else {
+                return false;
+            }
+
+        }
+
+        return isValid;
+    }
+
+    protected static boolean checkIfExtNum(String extKey) {
```

Comment:
should change this to checkIfExtRequiresQuotes and add BOOL_SUFFIX to the list below and return *false* (NOT *true*) for int/float/bool